pkg/network/placement_configuration.go

```diff
@@ -0,0 +1,54 @@
+package network
+
+import (
+	"reflect"
+
+	cnao "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/shared"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func fillDefaultsPlacementConfiguration(conf, previous *cnao.NetworkAddonsConfigSpec) []error {
+	if conf.PlacementConfiguration == nil || conf.PlacementConfiguration.Infra == nil || conf.PlacementConfiguration.Workloads == nil {
+		if previous != nil && previous.PlacementConfiguration != nil {
+			conf.PlacementConfiguration = previous.PlacementConfiguration
+			return []error{}
+		}
+
+		conf.PlacementConfiguration = &cnao.PlacementConfiguration{
+			Infra: &cnao.Placement{
+				NodeSelector: map[string]string{
+					"beta.kubernetes.io/arch":        "amd64",
+					"node-role.kubernetes.io/master": "",
+				},
+				Tolerations: []corev1.Toleration{
+					corev1.Toleration{
+						Key:      "node-role.kubernetes.io/master",
+						Operator: corev1.TolerationOpExists,
+						Effect:   corev1.TaintEffectNoSchedule,
+					},
+				},
+			},
+			Workloads: &cnao.Placement{
+				NodeSelector: map[string]string{
+					"beta.kubernetes.io/arch": "amd64",
+				},
+				Tolerations: []corev1.Toleration{
+					corev1.Toleration{
+						Operator: corev1.TolerationOpExists,
```

Comment:
shouldn't there be a `Key:      "node-role.kubernetes.io/worker",` or something the like?

Reply:
if that's the reason, then I think I prefer it to be explicit. is there a reason not to be explicit here?

Reply:
@phoracek pointed out that worker nodes don't have `node-role.kubernetes.io/worker` in vanilla kubernetes, despite this label is broadly used, we should rely on it. So that's why workloads are allowed to schedule on all nodes